Itm 613 rq filters

kaitlyn-sharo opened the pull request:
Added filters to all RQ tables. Filters are multiselect and include a textbox feature to search through filter options. 

Also added delegator_mil parameter to RQ1/RQ3 (including updating the variable definition table and download)

nextcen-dgemoets commented:
ZOMG I love this!!

Two very optional suggestions:
- Somewhere near the filters (either next to or below or something), add "Showing X of Y rows based on filters," or something like that.
- Since Download Data downloads all data regardless of filters, maybe change title to "Download All Data" or "Download Unfiltered Data".  Or even have two buttons ("Download Filtered Data" and "Download All Data").  Or one "Download Data" button, but add a nearby checkbox, "Filtered Data Only".  Might not have space for all of this-- all of this optional and just food for thought.

kaitlyn-sharo commented:
> ZOMG I love this!!
> 
> Two very optional suggestions:
> 
> * Somewhere near the filters (either next to or below or something), add "Showing X of Y rows based on filters," or something like that.
> * Since Download Data downloads all data regardless of filters, maybe change title to "Download All Data" or "Download Unfiltered Data".  Or even have two buttons ("Download Filtered Data" and "Download All Data").  Or one "Download Data" button, but add a nearby checkbox, "Filtered Data Only".  Might not have space for all of this-- all of this optional and just food for thought.

I like it! I added the "showing x of y rows" thing above the filters, only if filters are active. I put it above since the filters can grow long sometimes and putting it below could look weird. 

For your second point, I just changed it to "Download All Data". I think that if there comes a need for downloading filtered data, I can add that, but there are a lot more pressing things to do right now and they've never asked for that before (probably because it's easy to do in excel with the download!), so I will pass on that for the moment

brianpippin commented:
> > ZOMG I love this!!
> > Two very optional suggestions:
> > 
> > * Somewhere near the filters (either next to or below or something), add "Showing X of Y rows based on filters," or something like that.
> > * Since Download Data downloads all data regardless of filters, maybe change title to "Download All Data" or "Download Unfiltered Data".  Or even have two buttons ("Download Filtered Data" and "Download All Data").  Or one "Download Data" button, but add a nearby checkbox, "Filtered Data Only".  Might not have space for all of this-- all of this optional and just food for thought.
> 
> I like it! I added the "showing x of y rows" thing above the filters, only if filters are active. I put it above since the filters can grow long sometimes and putting it below could look weird.
> 
> For your second point, I just changed it to "Download All Data". I think that if there comes a need for downloading filtered data, I can add that, but there are a lot more pressing things to do right now and they've never asked for that before (probably because it's easy to do in excel with the download!), so I will pass on that for the moment

Agreed, we don't need the download filtered data at this time although a nice request that could be added to the backlog.

kaitlyn-sharo commented:
> > > ZOMG I love this!!
> > > Two very optional suggestions:
> > > 
> > > * Somewhere near the filters (either next to or below or something), add "Showing X of Y rows based on filters," or something like that.
> > > * Since Download Data downloads all data regardless of filters, maybe change title to "Download All Data" or "Download Unfiltered Data".  Or even have two buttons ("Download Filtered Data" and "Download All Data").  Or one "Download Data" button, but add a nearby checkbox, "Filtered Data Only".  Might not have space for all of this-- all of this optional and just food for thought.
> > 
> > 
> > I like it! I added the "showing x of y rows" thing above the filters, only if filters are active. I put it above since the filters can grow long sometimes and putting it below could look weird.
> > For your second point, I just changed it to "Download All Data". I think that if there comes a need for downloading filtered data, I can add that, but there are a lot more pressing things to do right now and they've never asked for that before (probably because it's easy to do in excel with the download!), so I will pass on that for the moment
> 
> Agreed, we don't need the download filtered data at this time although a nice request that could be added to the backlog.

https://nextcentury.atlassian.net/browse/ITM-618?atlOrigin=eyJpIjoiMThiODA3MGY2NzQ3NDZmZDg2MGMwMjU1YzNkMDExMTAiLCJwIjoiaiJ9

nextcen-dgemoets commented:
> I like it! I added the "showing x of y rows" thing above the filters, only if filters are active. I put it above since the filters can grow long sometimes and putting it below could look weird.
> 
> For your second point, I just changed it to "Download All Data". I think that if there comes a need for downloading filtered data, I can add that, but there are a lot more pressing things to do right now and they've never asked for that before (probably because it's easy to do in excel with the download!), so I will pass on that for the moment

Completely reasonable.  And the new "showing x of y rows" looks good.
